check/src/unify_type.rs

```diff
@@ -133,11 +133,8 @@ impl<I> Substitutable for ArcType<I> {
     type Variable = TypeVariable;
 
     fn new(id: u32) -> ArcType<I> {
-        Type::variable(TypeVariable::new(id))
-    }
-
-    fn from_variable(var: TypeVariable) -> ArcType<I> {
-        Type::variable(var)
+        // FIXME: using `Kind::hole()` here results in test failures
+        Type::variable(TypeVariable::new(id, Kind::typ()))
```

Comment:
This should be row https://github.com/gluon-lang/gluon/blob/9adf8be8065ebeb385c94829ad2368c307c95d6e/check/src/typecheck.rs#L607
https://github.com/gluon-lang/gluon/blob/9adf8be8065ebeb385c94829ad2368c307c95d6e/check/src/typecheck.rs#L766

Kind::Hole ? https://github.com/gluon-lang/gluon/blob/9adf8be8065ebeb385c94829ad2368c307c95d6e/check/src/typecheck.rs#L1199

generic.kind https://github.com/gluon-lang/gluon/blob/9adf8be8065ebeb385c94829ad2368c307c95d6e/check/src/typecheck.rs#L1288